packages/core/src/common/resource.ts

```diff
@@ -388,11 +410,3 @@ export class UntitledResource implements Resource {
         return undefined;
     }
 }
-
-export function createUntitledURI(extension?: string, parent?: URI): URI {
-    const name = `Untitled-${untitledResourceSequenceIndex++}${extension ?? ''}`;
-    if (parent) {
-        return parent.resolve(name).withScheme(UNTITLED_SCHEME);
-    }
-    return new URI().resolve(name).withScheme(UNTITLED_SCHEME);
-}
```

Comment:
Suggestion: Instead of deleting the function completely, we might want to deprecate it instead, to give downstream users a clear way for migration:

```ts
/**
 * @deprecated Since 1.27.0. Please use `UntitledResourceResolver.createUntitledURI` instead.
 */
export function createUntitledURI(extension?: string, parent?: URI): URI {
...
```